lib/Sema/CSSimplify.cpp

```diff
@@ -5984,14 +6009,118 @@ ConstraintSystem::simplifyCheckedCastConstraint(
       break;
   } while (true);
 
+  auto fixForCheckedConvertibleTypes = [&]() -> ConstraintFix * {
```

Comment:
I wonder if this could be split into `attempt/check` methods on different kinds of fixes being attempted here instead of a gigantic closure?

Reply:
I could abstract some parts but let the common checks such as earlier exits for "don't need to attempt any fix". WDYT?

Reply:
I did abstract some logic there, so its a bit less code here and most of then common to all fixes. So let me know what you think :)